### DISCUSSION TITLE
Replace MIT license with Boost license

### DISCUSSION THREAD

**Discussion by JasonCoombs** [NilFoundation/crypto3#20] [Ideas]
Using the MIT license for modules built from and extending Boost causes confusion.

It would be more clear exactly what the license restrictions are by remaining consistent and applying the Boost license instead.

e.g. Why do we not see crypto3-algebra in the Boost-ified version of crypto3?

https://github.com/nilfoundation/boost-crypto3

also, FYI, regarding boost-crypto3 on the nil.foundation website:

https://crypto3.nil.foundation/projects/crypto3/

there is no "develop" branch of boost-crypto3 as stated there.

https://github.com/NilFoundation/boost-crypto3/branches/all

**Comment by nemothenoone** [comment]
Hey @JasonCoombs! 

Thanks for noticing! The only module of ours which was directly inherited from Boost is [Crypto3.Multiprecision](https://github.com/nilfoundation/crypto3-multiprecision) and it keeps the same license as the original library. Others were either designed from scratch (e.g. [Crypto3.Algebra](https://github.com/nilfoundation/crypto3-algebra)) either reworked completely so there is nothing similar with the inspiration source anymore. So, we figured, this should create no license issues.

Regarding the Boost-tified version of Crypto3 suite. This version was prepared for the inclusion in Boost as a very basic subset of the original suite just to have something to start with. We hoped to move forward with more modules coming (e.g. putting [Crypto3.Algebra}(https://github.com/nilfoundation/crypto3-algebra) or [Crypto3.Pubkey](https://github.com/nilfoundation/crypto3-pubkey) in there step by step). But for now the process didn't go too much forward because of the absence of those who could audit this kind of things in Boost community. So we decided to step back for a while and proceed with the original suite 'till it becomes used widely enough for the audit to start seem being  redundant.

So, if you want to use the suite (amount of its' modules big enough), it would make sense to just submodule the original suite (https://github.com/nilfoundation/crypto3), add such a submodule as a subdirectory in CMake projects of yours, and link the CMake target of yours to the module you're interested in (e.g. `crypto3::algebra` or `crypto3::multiprecision` or `crypto3::pubkey` with something like `target_link_libraries(your_target_name PUBLIC crypto3::algebra)`). This should work well.

P.S. Converted this to the discussion inside the umbrella repository since this probably is about more than just Crypto3.Algebra.
